frontend/src/metabase/query_builder/components/view/ViewHeader.jsx

```diff
@@ -261,6 +262,19 @@ export class ViewTitleHeader extends React.Component {
               />
             </Box>
           )}
+          {question.query().database() && isNative && isSaved && (
+            <Link
+              to={question
+                .composeThisQuery()
+                .setDisplay("table")
+                .setSettings({})
+                .getUrl()}
+            >
+              <ViewButton medium p={[2, 1]} icon="insight" labelBreakpoint="sm">
+                {t`Explore results`}
+              </ViewButton>
+            </Link>
+          )}
           {isRunnable && !isNativeEditorOpen && (
```

Comment:
A bit out of context here, but it looks like those methods mutate the question. So once the `Link` is rendered, the question instance calls those methods and its state mutates without any action from a user. Is it the desired behavior? I see in the previous PR we were generating a new question, so the original one is not changed

Reply:
Good point! But I could not make it lead to any unexpected behavior

Reply:
Yeah interesting. I'm not super familiar with all the complexities of the query lib, so I was just going off of @paulrosenzweig's recommendation on the last PR. I haven't seen anything unexpected happen once the button shows up, and indeed calling `window.question.card().visualization_settings` after the button loads shows me the existing viz settings object not an empty object (I'd think it'd be a blank object given the `setSettings({})` call), which makes me think nothing about the state of the question is actually getting changed as far as the actual on screen question goes. Happy to switch back to the explicit `new Question` approach if we think that'd be safer, although I'm not sure if that'd come with its own potential downsides or loss of info.

Reply:
They don't mutate the question, but good thought to check! That would be a really annoying bug to track down.

[`setCard`](https://github.com/metabase/metabase/blob/dea5093096e4565ffa0e063d7b99c96bc1c2641d/frontend/src/metabase-lib/lib/Question.js#L182) clones the question before replace its card. Though, I guess its name seems to imply an update.